diff --git a/configure/os/CONFIG.Common.RTEMS b/configure/os/CONFIG.Common.RTEMS
index c8729f933..d0479e8b9 100644
--- a/configure/os/CONFIG.Common.RTEMS
+++ b/configure/os/CONFIG.Common.RTEMS
@@ -38,6 +38,12 @@ include $(RTEMS_BASE)/$(GNU_TARGET)$(RTEMS_VERSION)/$(RTEMS_BSP)/Makefile.inc
 include $(RTEMS_CUSTOM)
 include $(CONFIG.CC)
 
+#-----------
+# Krampf
+#not set in RTEMS_BSD.cfg until now
+RTEMS_HAS_NETWORKING = no
+RTEMS_HAS_POSIX_API = yes
+
 #-------------------------------------------------------
 # RTEMS cross-development tools
 CC = $(RTEMS_TOOLS)/bin/$(CC_FOR_TARGET) $(GCCSPECS) -fasm
@@ -115,7 +121,9 @@ OP_SYS_LDLIBS_score_NET_yes = -lnfs
 OP_SYS_LDLIBS_score_NET_no = -lnfs
 OP_SYS_LDLIBS += -lrtemsCom -lCom
 OP_SYS_LDLIBS += $(OP_SYS_LDLIBS_$(OS_API)_NET_$(RTEMS_HAS_NETWORKING))
+ifneq ($(RTEMS_VERSION),6)
 OP_SYS_LDLIBS += -lrtemscpu -lc -lm
+endif
 
 OP_SYS_LDFLAGS_posix = -u POSIX_Init
 OP_SYS_LDFLAGS_score = -u Init \
